src/components/mols/CreditCardPayment/index.tsx

```diff
@@ -0,0 +1,231 @@
+"use client";
+import React, { useState } from "react";
+import { Formik, Form, Field } from "formik";
+import { Input, ButtonPrimary, FormItem, Select } from "@/components";
+import { ICreditCardPayment } from "@/types/payment";
+import * as Yup from "yup";
+import valid from "card-validator";
+import Image from "next/image";
+import CardFlag from "@/images/easytolive/payment/card-flag.svg";
+import { MONTHS } from "@/constants/months";
+
+interface ICreditCardPaymentProps {
+  currentStepPayment: React.Dispatch<React.SetStateAction<number>>;
+}
+
+const CreditCardPayment: React.FC<ICreditCardPaymentProps> = ({
+  currentStepPayment,
+}) => {
+  const [loading, setLoading] = useState(false);
+  const [formattedCardNumber, setFormattedCardNumber] = useState("");
+
+  const CreditCardvalidationSchema = Yup.object().shape({
+    cardNumber: Yup.string()
+      .test(
+        "test-number",
+        "Número de cartão inválido",
+        (value) => valid.number(value).isValid,
+      )
+      .required("Número de cartão inválido"),
+    cvv: Yup.string().test(
+      "test-cvv",
+      "Verifique a validade do cartão ou CVV",
+      (value) => valid.cvv(value).isValid,
+    ),
+    nameOnCard: Yup.string()
+      .required("Nome do titular inválido")
+      .test(
+        "test-name",
+        "Nome do titular inválido",
+        (value) => valid.cardholderName(value).isValid,
+      ),
+    cardMonth: Yup.string()
+      .required()
+      .test(
+        "is-expired",
+        "Verifique a validade do cartão ou CVV",
+        function (value) {
+          const currentYear = new Date().getFullYear();
+          const currentMonth = new Date().getMonth() + 1;
+
+          if (
+            parseInt(value) < currentMonth &&
+            parseInt(this.parent.cardYear) <= currentYear
+          ) {
+            return false;
+          }
+
+          return true;
+        },
```

Comment:
Mover isso pra um util e chamar só: `isCreditCardExpirationValid(month, year)`

Reply:
Criei o util mas quando eu o chamo ai dentro ele lança o erro: Argument of type 'boolean' is not assignable to parameter of type 'TestFunction<string, AnyObject>'.ts(2345)
Pesquisei sobre e isso é por conta do "test" ser de uma biblioteca e ele só pode ser substituído por uma função que chama um elemento e eu substituí por uma que precisa de 2 argumentos e gera esse erro. Eu Conseguiria usando só um value, mas ai teria que conseguir o cardYaer que alguma outra forma que não acho viável.

Reply:
Atualizei la